devtools/maven/src/main/java/io/quarkus/maven/GenerateExtensionsJsonMojo.java

```diff
@@ -0,0 +1,203 @@
+            return;
+        }
+
+        final JsonArrayBuilder extListJson = Json.createArrayBuilder();
+        for (Dependency dep : deps) {
+            final Artifact artifact = dep.getArtifact();
+            if (!artifact.getExtension().equals("jar")
+                    || "javadoc".equals(artifact.getClassifier())
+                    || "tests".equals(artifact.getClassifier())
+                    || "sources".equals(artifact.getClassifier())) {
+                continue;
+            }
+            ArtifactResult resolved = null;
+            try {
+                resolved = repoSystem.resolveArtifact(repoSession,
+                        new ArtifactRequest().setRepositories(repos).setArtifact(artifact));
+            } catch (ArtifactResolutionException e) {
+                // there are some parent poms that appear as jars for some reason
+                debug("Failed to resolve dependency %s defined in %s", artifact, bomArtifact);
+            }
+            if (resolved != null) {
+                processDependency(resolved.getArtifact(), extListJson);
+            }
+        }
+
+        final JsonObjectBuilder platformJson = Json.createObjectBuilder();
+        final JsonObjectBuilder bomJson = Json.createObjectBuilder();
+        bomJson.add("groupId", bomGroupId);
+        bomJson.add("artifactId", bomArtifactId);
+        bomJson.add("version", bomVersion);
+        platformJson.add("bom", bomJson.build());
+        platformJson.add("extensions", extListJson.build());
+
+        final File outputDir = outputFile.getParentFile();
+        if (!outputDir.exists()) {
+            if (!outputDir.mkdirs()) {
+                throw new MojoExecutionException("Failed to create output directory " + outputDir);
+            }
+        }
+        final JsonWriterFactory jsonWriterFactory = Json
+                .createWriterFactory(Collections.singletonMap(JsonGenerator.PRETTY_PRINTING, true));
+        try (JsonWriter jsonWriter = jsonWriterFactory.createWriter(new FileOutputStream(outputFile))) {
+            jsonWriter.writeObject(platformJson.build());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to persist " + outputFile, e);
+        }
+
+        projectHelper.attachArtifact(project, "json", null, outputFile);
+    }
+
+    private void processDependency(Artifact artifact, JsonArrayBuilder extJsonBuilder) {
+        final Path path = artifact.getFile().toPath();
+        try {
+            if (Files.isDirectory(path)) {
+                processMetaInfDir(artifact, path.resolve(BootstrapConstants.META_INF), extJsonBuilder);
+            } else {
+                try (FileSystem artifactFs = ZipUtils.newFileSystem(path)) {
+                    processMetaInfDir(artifact, artifactFs.getPath(BootstrapConstants.META_INF), extJsonBuilder);
+                }
+            }
+        } catch (Throwable t) {
```

Comment:
shouldn't this not just catch (Exception e) ?

Reply:
Thanks, I am actually throwing a wrong exception here. Copy-pasted from another class.